### PR TITLE
Fixed HA warning

### PR DESCRIPTION
**Pull request by timlaing**
Fixes HA warnings

*Detected that custom integration 'pod_point' calls `async_add_job`, which is deprecated and will be removed in Home Assistant 2025.4; Please review https://developers.home-assistant.io/blog/2024/03/13/deprecate_add_run_job

* Detected code that calls async_forward_entry_setup for integration, pod_point with title: <> and entry_id: <>, which is deprecated and will stop working in Home Assistant 2025.6, await async_forward_entry_setups instead

* Detected that custom integration 'pod_point' calls hass.http.register_static_path which is deprecated because it does blocking I/O in the event loop, instead call `await hass.http.async_register_static_paths([StaticPathConfig("/api/pod_point/static", "/config/custom_components/pod_point/static", False)])`; This function will be removed in 2025.7 at custom_components/pod_point/__init__.py, line 92: hass.http.register_static_path(, please create a bug report at https://github.com/mattrayner/pod-point-home-assistant-component/issues

### CODE REVIEW

**Review by mattrayner — DISMISSED** [review]
This looks like a great contribution, I'll test it in the morning but thanks so much!

**Comment by timlaing** [comment]
I have added updated to fix dependancy issues and unit tests

**Comment by timlaing** [comment]
@mattrayner - I'll take a look at why the workflow checks have failed tomorrow and push an update once I have found and fixed the errors.

**Comment by mattrayner** [comment]
@timlaing if you can focus on the tests failing I'll update the hassfest complaints to resolve those in a seperate PR, so happy to merge this conce the tests are green again

**Comment by timlaing** [comment]
@mattrayner - I have fixed all the errors for you.

File: custom_components/pod_point/manifest.json [hunk 1]
```diff
@@ -563,7 +564,7 @@
   "integration_type": "hub",
   "iot_class": "cloud_polling",
   "issue_tracker": "https://github.com/mattrayner/pod-point-home-assistant-component/issues",
-  "quality_scale": "gold",
+  "quality_scale": "silver",
```
Comment (mattrayner) [inline custom_components/pod_point/manifest.json:567]: I'm going to open a PR to add the additional diagnostics entity

**Review by mattrayner — APPROVED** [review]
LGTM

**Pull request merged**